devicetypes/smartthings/fidure-2nd-generation-thermostat.src/fidure-2nd-generation-thermostat.groovy

```diff
@@ -0,0 +1,810 @@
+	
+	cmds += setThermostatTime()
+	
+	sendZigbeeCmds(cmds, 100)
+
+	runIn(5, "refresh", [overwrite: true]) 
+}
+def setDefaultMinMax(){
+	device.updateDataValue("minheatSetpoint", "830") 
+	device.updateDataValue("maxheatSetpoint", "3000") 
+	device.updateDataValue("mincoolSetpoint", "1500") 
+	device.updateDataValue("maxcoolSetpoint", "3780") 
+}
+
+def	turnProgOff() {	zigbee.writeAttribute(0x0201,0x0025, 0x18, 0) }
+def	turnProgOn() { zigbee.writeAttribute(0x0201, 0x0025, 0x18, 1)}
+def	turnHoldOn() { zigbee.writeAttribute(0x0201, 0x0023 , 0x30 , 1)}
+def turnHoldOff() {zigbee.writeAttribute(0x0201, 0x0023 , 0x30 , 0)}
+
+
+def parse(String description) {
+	Map[] maps = []
+	Map map = [:]
+	def evts = []
+
+	log.debug "got parse: $description"
+	sendEvent("name":"lastSeen", "value":  (new Date()).format("MM-dd HH:mm",location.timeZone) )
+
+	//POSSIBLE ST BUG: for some reason humidity reports show up already parsed. but the % sign causes an error
+	if (description?.startsWith('humidity:')){ 
+		map.name = "humidity"
+		map.value = description.split(":")[1] - "%"
+	} else 
+		map = zigbee.getEvent(description)
+
+	if (!map)
+	{
+		if (description?.startsWith('catchall:')) {
+			maps += parseCatchAllMessage(description)
+		} else if (description?.startsWith('read attr -')) {
+		maps += parseReportAttributeMessage(description)
+		}
+	} else maps += map
+
+	log.debug "Parse returned $maps"
+	maps.each { parsedMap ->
+		evts += parsedMap ? createEvent(parsedMap) : [:]
+    }
+
+	// check clock sync
+	getTimeAndDay()
+
+	return evts
+}
+
+private getTimeAndDay() {
+	def timeNow = now()
+	// Need to check that location have timeZone as SC may have created the location without setting it
+	// update clock every 4 hours (for units that don't have an RTC)
+	if (location.timeZone && (!state.timeClockSet || (4 * 60 * 60 * 1000 < (timeNow - state.timeClockSet)))) {
+		state.timeClockSet = timeNow
+		log.debug "updating time: $timeNow"
+		updateTime();
+	}
+}
+
+private Map[] parseMultipleAttributes(String cluster, String rawZigBee) {
+	Map[] result = []
+	Map descMap = [:]
+	Integer size
+
+	while(rawZigBee.length() >= 8)
+	{
+		descMap["cluster"] = cluster
+		descMap["attrId"] = reverseHex(rawZigBee.substring(0,4))
+		rawZigBee = rawZigBee.substring(4)
+		descMap["encoding"] = rawZigBee.substring(0,2)
+		rawZigBee = rawZigBee.substring(2)
+		size = Integer.parseInt("${descMap.encoding}", 16)
+		size = DataType.getLength(size) * 2 // hex
+		descMap["value"] = reverseHex(rawZigBee.substring(0, size))
+		rawZigBee = rawZigBee.substring(size)
+		result += descMap
+		descMap = [:]
+	}
+	log.debug "multi attribute decode: $result"
+	return result
+}
+
+private Map processSingleAttributeMessge(Map descMap){
+	
+	Map resultMap = [:]
+	if (descMap.cluster == "0201") {
+		resultMap = parseThermostatClusterAttr(descMap)
+	}
+	else if (descMap.cluster == "0405" && descMap.attrId == "0000") {
+		resultMap.name = "humidity"
```

Comment:
if not handled anywhere else, make sure all relevant events include the correct unit